---
title: "Annotating metApeakFisher_joint_peaks_18lcls"
author: Yanyu Liang
date: "`r format(Sys.time(), '%d %B, %Y')`"
---

# Dependencies

```{r}
library(ggplot2)
library(dplyr)
```

# Load Data

```{r}
mydat <- read.table('../analysis/transcript_annotation/summary/metApeakFisher_joint_peaks_18lcls.tab.gz',
                    sep = '\t',
                    header = T)
```

# Extract valid transcript hit

This is defined as the peak-transcript hit in which the peak locates completely in the transcript (namely exon).

```{r}
mydat.valid <- mydat[mydat$size == mydat$exon, ]
if(length(unique(mydat.valid$peak.id)) != length(unique(mydat$peak.id)))  {  # To check if all peak locate in at least one transcript
  print('warning: Some peaks do NOT locate comppletely in transcript. They are: ')
  peaks.all <- unique(mydat.valid$peak.id)
  print(peaks.all[!peaks.all %in% mydat.valid$peak.id])
}   
```

# Number of valid transcripts per peak

```{r}
hit <- mydat.valid$peak.id
valid.count <- data.frame(table(hit))
ggplot(valid.count) + geom_histogram(aes(x = Freq), binwidth = 1) + ggtitle('# of valid transcript per peak')
```

# Do they locate in exon?

```{r}
ggplot(mydat.valid) + geom_point(aes(x = size, y = exon)) +
  geom_abline(slope = 1, intercept = 0) +
  ggtitle('Peak size versus the size of annotated exon')
```

# Are they protein coding genes (PCGs)?

```{r}
ggplot(mydat.valid) +
  geom_point(aes(x = exon, y = CDS + fiveUTR + threeUTR)) + geom_abline(slope = 1, intercept = 0) +
  ggtitle('Annotated exon versus CDS + UTRs') + labs(y = 'CDS + UTRs')
```

# Where do PCG peaks locate?

```{r}
mydat.pcg <- mydat.valid %>%
  filter(CDS + threeUTR + fiveUTR != 0)
ggplot(mydat.pcg) +
  geom_point(aes(x = fiveUTR / size, y = threeUTR / size, color = CDS / size > 0.75)) +
  # scale_color_gradient2(low = 'blue', mid = 'white', high = 'red', midpoint = 0.75,  limits=c(0, 1)) +
  ggtitle('Fraction of UTRs/CDS in PCG peaks') +
  labs(x = 'Fraction of 5\'UTR', y = 'Fraction of 3\'UTR', color = 'CDS % > 0.75') +
  annotate("rect", xmin = 0.75, xmax = 1, ymin = 0, ymax = 0.1,
  alpha = .2) +
  annotate("rect", xmin = 0, xmax = 0.1, ymin = 0.75, ymax = 1,
  alpha = .2)
```
