---
title: "Annotating metApeakFisher_joint_peaks_18lcls"
author: Yanyu Liang
date: "`r format(Sys.time(), '%d %B, %Y')`"
---

# Dependencies

```{r}
library(ggplot2)
library(dplyr)
source('../code/utilities/transcript_annotation_rlib.R')
```

# Load Data

```{r}
mydat <- read.table('../analysis/transcript_annotation/summary/metApeakFisher_joint_peaks_18lcls.tab.gz',
                    sep = '\t',
                    header = T)
```

# Extract valid transcript hit

This is defined as the peak-transcript hit in which the peak locates completely in the transcript (namely exon).

```{r}
mydat.valid <- mydat[mydat$size == mydat$exon, ]
if(length(unique(mydat.valid$peak.id)) != length(unique(mydat$peak.id)))  {  # To check if all peak locate in at least one transcript
  print('warning: Some peaks do NOT locate comppletely in transcript. They are: ')
  peaks.all <- unique(mydat$peak.id)
  print(peaks.all[!peaks.all %in% mydat.valid$peak.id])
}
```

# Number of valid transcripts per peak

```{r}
output <- countFreq(mydat.valid)
mydat.valid.count <- output$df
valid.count <- output$count
ggplot(valid.count) + geom_histogram(aes(x = Freq), binwidth = 1) + ggtitle('# of valid transcript per peak') + scale_y_log10()
```

# Do they locate in exon?

```{r}
ggplot(mydat.valid.count) + geom_jitter(aes(x = size, y = exon, color = log2.freq), width = 200) +
  geom_abline(slope = 1, intercept = 0) +
  ggtitle('Peak size versus the size of annotated exon') +
  scale_color_gradient2(low = 'blue', mid = 'white', high = 'red', midpoint = 2)
```

# Are they protein coding genes (PCGs)?

```{r}
ggplot(mydat.valid.count) +
  geom_jitter(aes(x = exon, y = CDS + fiveUTR + threeUTR, color = log2.freq), width = 200) + geom_abline(slope = 1, intercept = 0) +
  ggtitle('Annotated exon versus CDS + UTRs') + labs(y = 'CDS + UTRs') +
  scale_color_gradient2(low = 'blue', mid = 'white', high = 'red', midpoint = 2)
```

# Where do PCG peaks locate?

```{r}
mydat.pcg <- mydat.valid %>%
  filter(CDS + threeUTR + fiveUTR != 0)
output <- countFreq(mydat.pcg)
mydat.pcg.count <- output$df
ggplot(mydat.pcg.count) +
  geom_jitter(aes(x = fiveUTR / size, y = threeUTR / size, color = log2.freq)) +
  # scale_color_gradient2(low = 'blue', mid = 'white', high = 'red', midpoint = 0.75,  limits=c(0, 1)) +
  ggtitle('Fraction of UTRs/CDS in PCG peaks') +
  labs(x = 'Fraction of 5\'UTR', y = 'Fraction of 3\'UTR') +
  annotate("rect", xmin = 0.75, xmax = 1, ymin = 0, ymax = 0.1,
  alpha = .2) +
  annotate("rect", xmin = 0, xmax = 0.1, ymin = 0.75, ymax = 1,
  alpha = .2)
```

# Inconsistency among transcript annotations

```{r}
mydat.pcg.dist <- mydat.pcg %>%
  group_by(peak.id) %>%
  summarise(distance = sd(fiveUTR/exon) + sd(threeUTR/exon)) %>%
  ungroup()
ggplot(mydat.pcg.dist) +
  geom_histogram(aes(x = distance)) +
  scale_y_log10() +
  ggtitle('total deviation of annotation 
 in single peak 
 within multiple transcripts')
```

# Stratifying peaks

```{r}
mydat.valid.stratify <- mydat.valid %>%
  mutate(type = stratifyMatch(fiveUTR, threeUTR, CDS, exon))
mydat.valid.stratify.bypeak <- mydat.valid.stratify %>%
  group_by(peak.id) %>%
  summarise(peak.type = stratifyPeak(type)) %>%
  ungroup()
ggplot(mydat.valid.stratify.bypeak) +
  geom_bar(aes(x = peak.type, y = ..count..)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle('Number of peak per annotation category')
```
