<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yanyu Liang" />


<title>Annotating metApeakFisher_joint_peaks_18lcls</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Deep M6A</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Results</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/liangyy/deep_m6a">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Annotating metApeakFisher_joint_peaks_18lcls</h1>
<h4 class="author"><em>Yanyu Liang</em></h4>
<h4 class="date"><em>27 September, 2017</em></h4>

</div>


<div id="dependencies" class="section level1">
<h1><span class="header-section-number">1</span> Dependencies</h1>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
</div>
<div id="load-data" class="section level1">
<h1><span class="header-section-number">2</span> Load Data</h1>
<pre class="r"><code>mydat &lt;- read.table(&#39;../analysis/transcript_annotation/summary/metApeakFisher_joint_peaks_18lcls.tab.gz&#39;,
                    sep = &#39;\t&#39;,
                    header = T)</code></pre>
</div>
<div id="extract-valid-transcript-hit" class="section level1">
<h1><span class="header-section-number">3</span> Extract valid transcript hit</h1>
<p>This is defined as the peak-transcript hit in which the peak locates completely in the transcript (namely exon).</p>
<pre class="r"><code>mydat.valid &lt;- mydat[mydat$size == mydat$exon, ]
if(length(unique(mydat.valid$peak.id)) != length(unique(mydat$peak.id)))  {  # To check if all peak locate in at least one transcript
  print(&#39;warning: Some peaks do NOT locate comppletely in transcript. They are: &#39;)
  peaks.all &lt;- unique(mydat.valid$peak.id)
  print(peaks.all[!peaks.all %in% mydat.valid$peak.id])
}   </code></pre>
<pre><code>## [1] &quot;warning: Some peaks do NOT locate comppletely in transcript. They are: &quot;
## factor(0)
## 18567 Levels: chr1 100038232 100049974 HIAT1 2 84,66, 0,11676 ...</code></pre>
</div>
<div id="number-of-valid-transcripts-per-peak" class="section level1">
<h1><span class="header-section-number">4</span> Number of valid transcripts per peak</h1>
<pre class="r"><code>hit &lt;- mydat.valid$peak.id
valid.count &lt;- data.frame(table(hit))
ggplot(valid.count) + geom_histogram(aes(x = Freq), binwidth = 1) + ggtitle(&#39;# of valid transcript per peak&#39;)</code></pre>
<p><img src="transcript_annotation.metApeakFisher_joint_peaks_18lcls_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="do-they-locate-in-exon" class="section level1">
<h1><span class="header-section-number">5</span> Do they locate in exon?</h1>
<pre class="r"><code>ggplot(mydat.valid) + geom_point(aes(x = size, y = exon)) +
  geom_abline(slope = 1, intercept = 0) +
  ggtitle(&#39;Peak size versus the size of annotated exon&#39;)</code></pre>
<p><img src="transcript_annotation.metApeakFisher_joint_peaks_18lcls_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="are-they-protein-coding-genes-pcgs" class="section level1">
<h1><span class="header-section-number">6</span> Are they protein coding genes (PCGs)?</h1>
<pre class="r"><code>ggplot(mydat.valid) +
  geom_point(aes(x = exon, y = CDS + fiveUTR + threeUTR)) + geom_abline(slope = 1, intercept = 0) +
  ggtitle(&#39;Annotated exon versus CDS + UTRs&#39;) + labs(y = &#39;CDS + UTRs&#39;)</code></pre>
<p><img src="transcript_annotation.metApeakFisher_joint_peaks_18lcls_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="where-do-pcg-peaks-locate" class="section level1">
<h1><span class="header-section-number">7</span> Where do PCG peaks locate?</h1>
<pre class="r"><code>mydat.pcg &lt;- mydat.valid %&gt;%
  filter(CDS + threeUTR + fiveUTR != 0)
ggplot(mydat.pcg) +
  geom_point(aes(x = fiveUTR / size, y = threeUTR / size, color = CDS / size &gt; 0.75)) +
  # scale_color_gradient2(low = &#39;blue&#39;, mid = &#39;white&#39;, high = &#39;red&#39;, midpoint = 0.75,  limits=c(0, 1)) +
  ggtitle(&#39;Fraction of UTRs/CDS in PCG peaks&#39;) +
  labs(x = &#39;Fraction of 5\&#39;UTR&#39;, y = &#39;Fraction of 3\&#39;UTR&#39;, color = &#39;CDS % &gt; 0.75&#39;) +
  annotate(&quot;rect&quot;, xmin = 0.75, xmax = 1, ymin = 0, ymax = 0.1,
  alpha = .2) +
  annotate(&quot;rect&quot;, xmin = 0, xmax = 0.1, ymin = 0.75, ymax = 1,
  alpha = .2)</code></pre>
<p><img src="transcript_annotation.metApeakFisher_joint_peaks_18lcls_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
